Helpers/CommonMenuItemPartMigrator.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Etch.OrchardCore.Menu.Models;
+using Etch.OrchardCore.Widgets.Models;
+using GraphQL;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Autoroute.Models;
+using OrchardCore.ContentFields.Fields;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.ContentManagement.Routing;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Menu.Models;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace Etch.OrchardCore.Menu.Helpers
+{
+    public class CommonMenuItemPartMigrator : ICommonMenuItemPartMigrator
+    {
+        private readonly AutorouteOptions _autorouteOptions;
+        private readonly IContentManager _contentManager;
+        private readonly ILogger<CommonMenuItemPartMigrator> _logger;
+        private readonly ISession _session;
+        private readonly ShellSettings _settings;
+
+        public CommonMenuItemPartMigrator(IOptions<AutorouteOptions> autorouteOptions,
+            IContentManager contentManager,
+            ILogger<CommonMenuItemPartMigrator> logger,
+            ISession session,
+            ShellSettings settings)
+        {
+            _autorouteOptions = autorouteOptions.Value;
+            _contentManager = contentManager;
+            _logger = logger;
+            _session = session;
+            _settings = settings;
+        }
+
+        public async Task GetCommonMenuitemDataAsync()
+        {
+            foreach (var contentItem in await FetchMenuItemsAsync())
+            {
+                var menuItemsListPart = await MenuItemMigratorAsync(contentItem.As<MenuItemsListPart>());
+
+                contentItem.Apply(nameof(MenuItemsListPart), menuItemsListPart);
+                await _contentManager.UpdateAsync(contentItem);
+                await _contentManager.PublishAsync(contentItem);
+            };
+        }
+
+        private async Task<string> ContentItemIdToUrlAsync(string contentItemId)
+        {
+            var contentItem = await _session.Query<ContentItem>()
+                .With<ContentItemIndex>()
+                    .Where(x => x.ContentItemId == contentItemId)
+                .FirstOrDefaultAsync();
+
+            if (contentItem == null)
+            {
+                return string.Empty;
+            }
+
+            if (contentItem.As<AutoroutePart>() != null)
+            {
+                return contentItem.As<AutoroutePart>().Path;
+            }
+
+            return string.Empty;

```

Comment:
If there is no `AutoroutePart` there is still a URL for a content item, it's something like `Content/Display/<contentItemId>`. Try creating a content type without an autoroute part and publishing a content item, that will tell you what the URL should be.

Reply:
so i created a type, added only a title part to it. 

when i go to view the page the url is - `https://localhost:44358/studio/Contents/ContentItems/44ctbdvfs2nyf12mx39v2qg9cd`

so safe to assume setting path as `/studio/Contents/ContentItems/contentItemId`

i can add '"Contents/ContentItems/contentItemId"' at the end of the path variable which adds the url prefix